src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -90,26 +83,27 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         }
 
         // dispatch to the correct handler
-        if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsBefore && 
+        if (objResult.getsObjectType() == Allocation__c.sObjectType && !ALLO_ProcessControl.checkRecursionState() &&
             (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate)) {
 
             runAllocationBeforeTriggerHandler(newlist, oldlist, triggerAction, objResult);
-            hasRunForAllocationsBefore = true;
-        } else if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsAfter && 
+        } else if (objResult.getsObjectType() == Allocation__c.sObjectType && !ALLO_ProcessControl.checkRecursionState() &&
```

Comment:
was the separation of Allocations Before and After flags not really needed?  i see you are now using the same flag/counter.